backend/wmg/api/v2.py

```diff
@@ -173,6 +184,7 @@ def markers():
     snapshot: WmgSnapshot = load_snapshot(
         snapshot_schema_version=WMG_API_SNAPSHOT_SCHEMA_VERSION,
         explicit_snapshot_id_to_load=WMG_API_FORCE_LOAD_SNAPSHOT_ID,
+        snapshot_local_disk_path=SNAPSHOT_LOCAL_DISK_PATH,
```

Comment:
TBH I'm not a huge fan of the required arguments for load_snapshot. Would it be possible to put these defaults in the kwargs of the function definition? That way we could run `load_snapshot()` without needing all the extra verbosity? I only bring this up here because we're adding yet another required argument. To replicate deployment's behavior in a local kernel now requires importing or hardcoding all these extra constants and it's a bit annoying.

Reply:
@atarashansky - Remember that `load_snapshot()` is used by the pipeline too. Setting `snapshot_local_disk_path=SNAPSHOT_LOCAL_DISK_PATH` as the in `def load_snapshot(...)` wouldn't make sense since we wouldn't want the pipeline run to be reading from local disk.

_We want to keep the API decoupled from the pipeline (decoupling the reader from the writer)_.

That is why the function declaration in `backend/wmg/data` looks like this:

```
def load_snapshot(
    *,
    snapshot_schema_version: str,
    explicit_snapshot_id_to_load: Optional[str] = None,
    snapshot_local_disk_path: Optional[str] = None,
) -> WmgSnapshot
```

I think specifying defaults in the function declaration that pin it to values for WMG API violates the decoupling between the API and the pipeline.

I understand your point about having to specify these values being annoying but there is a straightforward fix for this:

Make wrapper functions for the reader and writer that wrap `load_snapshot(...)`. These wrapper functions would have sensible defaults for the API and pipeline and call pass them the call to `load_snapshot(...)`.

```
# WMG API

def api_load_snapshot(
    snapshot_schema_version=WMG_API_SNAPSHOT_SCHEMA_VERSION,
    explicit_snapshot_id_to_load=WMG_API_FORCE_LOAD_SNAPSHOT_ID,
    snapshot_local_disk_path=SNAPSHOT_LOCAL_DISK_PATH):

    return load_snapshot(...)
```

```
# WMG Pipeline

def pipeline_load_snapshot(snapshot_schema_version=WMG_DATA_SCHEMA_VERSION):
    return load_snapshot(...)
```

This way, API component callers can call `api_load_snapshot()` without ever having to supply arguments since there will always sensible defaults going forward. Similarly, pipeline component callers can call `pipeline_load_snapshot()` without supplying arguments 

If you buy my reasoning and this approach, I propose we do this in another PR rather than in this PR because several call sites would need to change.

Reply:
We can add this in a separate PR. I'm onboard with the proposed solution!